pints/_mcmc/__init__.py

```diff
@@ -311,14 +311,23 @@ def __init__(self, log_pdf, chains, x0, sigma0=None, method=None):
             raise ValueError('Number of chains must be at least 1.')
 
         # Check initial position(s): Most checking is done by samplers!
-        if len(x0) != chains:
-            raise ValueError(
-                'Number of initial positions must be equal to number of'
-                ' chains.')
-        if not all([len(x) == self._n_parameters for x in x0]):
-            raise ValueError(
-                'All initial positions must have the same dimension as the'
-                ' given LogPDF.')
+        self._x0_isfunction = hasattr(x0, '__call__')
+        if self._x0_isfunction:
```

Comment:
I think you can use `if callable(x0):`

Reply:
Or even just check if it's a LogPrior, otherwise assume it's a number

Reply:
Yep, good point. Have done the latter.